src/utils/NumberUtils.ts

```diff
@@ -15,6 +15,19 @@ export class NumberUtils {
         return `${Utils.hexadecimalPrefix}${basePart}`;
     }
 
+    /**
+     * @param {number} number
+     * @returns {[number, (number | null)]}
+     */
+    public static extractIntegerAndDecimalParts (number: number): [number, number | null] {
+        const integerPart: number = Math.floor(number);
```

Comment:
I think this line should be `Math.trunc(number)` instead of floor: otherwise for `number = -1e-100` you would get `[-1, -1e-100]`.
(I heard that Math.trunc doesn't work for IE; if that's an issue, maybe use `number >= 0 ? Math.floor : Math.ceil`)